Sources/Datadog/OpenTracing/OTSpan.swift

```diff
@@ -91,18 +91,21 @@ public extension OTSpan {
     /// - parameter error: An object conforming to the `Error` protocol.
     /// - parameter file: A string identifying the file where the `Error` was caught. The default is `#fileID` which means `ModuleName/Filename.extension`, consider an helpful yet concise identifier when overriding the default.
     /// - parameter line: The line number in the file where the `Error` was caught.
+    /// - parameter includeFileInStack: Whether or not the `file` and `line` should be included in the stack, the default is `true` and can be overriden if the `file`/`line` are extraneous.
     func setError(
         _ error: Error,
         file: StaticString = #fileID,
-        line: UInt = #line
+        line: UInt = #line,
+        includeFileInStack: Bool = true
```

Comment:
I'm not a huge fan of [flag arguments], especially in the public API if we add it only for our own need. I see 2 options to get rid of it:

1/ Add `includeFileInStack` in `internal` API and call it with `false` from our code and `true` from `public` API.

2/ Remove this argument, force the `file:` and `line:` values in our custom code with `#sourceLocation`:
```swift
// URLSessionTracingHandler.swift

#sourceLocation(file: "", line: 0)
span.setError(error)
#sourceLocation()
```
and skip it here in `OTSpan` implementation if the `file` is `""`.

WDYT?

Reply:
Yes I agree that it's not a good path to take for a public API and I previously considered the option #2 but that means assuming a rule (i.e. `file.count == 0` means skip it), it would be implicit as far as the API is concerned and I'm not sure that this is sound either.

I like the option 1 better, it keeps things clear at the public/internal API levels while going through the same code path.

An alternative that I have considered was to expect an Optional-like enum instead of a `StaticString` that defaults to `#fileID`, but I discarded it because it was too complex... Something that looked like `line: .none` or `line: .some("...")` and default `line: .some(#filedID)`. 

I think the more sensible option is 1 at this point. Thanks for suggesting this!

Reply:
Well, don't mind my previous hesitation, I already included a similar choice as far as the `stack` is concerned (skipped if empty)!
Option 2 shouldn't be a problem after all :)